### PR TITLE
rust/clippy: disable redundant_clone

### PR DESCRIPTION
**Pull request by benma**
Has false positives that can be very subtle, so best to play it
safe.

https://github.com/async-rs/async-task/issues/23

### CODE REVIEW

File: src/rust/bitbox02-rust-c/src/bitboxbase.rs [hunk 1]
```diff
@@ -207,7 +207,7 @@ pub extern "C" fn bitboxbase_state_set_not_alive() {
 #[no_mangle]
 pub extern "C" fn bitboxbase_state_get() -> BitBoxBaseBackgroundState {
     let state = unsafe { &STATE };
-    state.state.clone()
+    state.state
```
Comment (benma) [inline src/rust/bitbox02-rust-c/src/bitboxbase.rs:210]: @NickeZ clippy warned this

https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

Sounds legit, do you remember why there was a clone() here?

**Comment by x1ddos** [comment]
Hm... I'm quite disappointed with clippy at this point. It's always best to leave uncertain things as is and avoid false positives.

**Comment by benma** [comment]
> Hm... I'm quite disappointed with clippy at this point. It's always best to leave uncertain things as is and avoid false positives.

agree, at least the default linters should be 100% false positive free, when it comes to semantics.

File: src/rust/bitbox02-rust-c/src/bitboxbase.rs (hunk 1 above)
Comment (NickeZ) [reply src/rust/bitbox02-rust-c/src/bitboxbase.rs:210]: Nope, can't remember. Usually I add `clone()`s as temporary fixes to make the borrowchecker happy. Having many clones is a bit of a code smell.

**Pull request merged**